include/h2o.h

```diff
@@ -962,6 +962,10 @@ struct st_h2o_req_t {
          * offset of '?' within path, or SIZE_MAX if not found
          */
         size_t query_at;
+        /*
+         * list of request headers
+         */
+        h2o_headers_t headers;
```

Comment:
Could you please clarify the intent of adding this in this PR?

Reply:
Before this PR, the request headers were never modified, so we can always use req->headers for logging (with `%{HEADERNAME}i` notations). But from this PR the users can modify it in mruby like `env['HTTP_USER_AGENT'] = 'another value'`, so I think we should preserve original request headers as other values (scheme, method, path, etc)